core/src/main/java/org/infinispan/util/logging/events/impl/EventLogManagerImpl.java

```diff
@@ -25,6 +39,6 @@ public EventLogger replaceEventLogger(EventLogger newLogger) {
 
    @Override
    public EventLogger getEventLogger() {
-      return logger;
+      return Objects.requireNonNull(logger, "Logger can not be null!");
```

Comment:
I think `Objects#requireNonNull` is unnecessary here as the initialisation path is controlled by our CDI code. If `@Start` doesn't work as expected, then we'll have many more failures that would prevent `#getEventLogger` ever being called.